nextclade/Snakefile

```diff
@@ -0,0 +1,375 @@
+            --fields {params.fields} \
+            --output-metadata {output.metadata} \
+            --output-sequences {output.sequences}
+        """
+
+
+def genes(w):
+    if w.segment=='ha': return ["SigPep", "HA1", "HA2"]
+    if w.segment=='na': return ["NA"]
+
+rule subsample:
+    input:
+        aligned_sequences=rules.parse.output.sequences,
+        enriched_metadata=rules.parse.output.metadata,
+        include_strains="../config/{lineage}/reference_strains.txt",
+        nextclade_include="dataset_config/{lineage}/includes.txt",
+        exclude="../config/{lineage}/outliers.txt",
+    output:
+        sampled_sequences="build/{lineage}/{segment}/{reference}/subsample.fasta",
+        sampled_strains="build/{lineage}/{segment}/{reference}/subsample.txt",
+    params:
+        filter_arguments=lambda w: config["builds"][w.lineage][w.segment]["refs"][
+            w.reference
+        ]["filter"],
+        reference_EPI_ISL=lambda w: config["builds"][w.lineage][w.segment]["refs"][
+            w.reference
+        ]["reference_EPI_ISL"]
+    shell:
+        """
+        augur filter \
+            --sequences {input.aligned_sequences} \
+            --metadata {input.enriched_metadata} \
+            --exclude {input.exclude} \
+            --include {input.include_strains} {input.nextclade_include} \
+            --include-where EPI_ISL={params.reference_EPI_ISL} \
+            {params.filter_arguments} \
+            --output {output.sampled_sequences} \
+            --output-strains {output.sampled_strains}
+        """
+
+rule align:
+    input:
+        sequences="build/{lineage}/{segment}/{reference}/subsample.fasta",
+        annotation="dataset_config/{lineage}/{segment}/{reference}/annotation.gff",
+        reference="dataset_config/{lineage}/{segment}/{reference}/reference.fasta",
+    output:
+        alignment="build/{lineage}/{segment}/{reference}/align.aligned.fasta"
+    params:
+        outdir=lambda w: f"build/{w.lineage}/{w.segment}/{w.reference}/aligned.gene.{{gene}}.fasta",
+        nextclade_bin = "./nextclade_v3"
+    threads: 3
+    shell:
+        """
+        {params.nextclade_bin} run \
+            --jobs={threads} \
+            --input-ref {input.reference} \
+            --input-annotation {input.annotation} \
+            --output-translations {params.outdir} \
+            --output-fasta {output.alignment} \
+            {input.sequences} \
+            2>&1
+        """
+
+
+rule tree:
+    input:
+        alignment=rules.align.output.alignment,
+    output:
+        tree="build/{lineage}/{segment}/{reference}/tree_raw.nwk",
+    threads: 3
+    shell:
+        """
+        augur tree \
+            --alignment {input.alignment} \
+            --output {output.tree} \
+            --nthreads {threads} \
+            --tree-builder-args '-ninit 10 -n 4 -czb' \
+            > /dev/null
+        """
+
+# root using dates in treetime, use 1500 as sequence length (good enough, doesn't matter)
+rule root:
```

Comment:
I find myself needing this kind of functionality for other projects, too, where I want to root the tree before refining it and often when I don't want a time tree. This is usually in the context of rooting on the same strain I used as the reference for alignment and I usually want to prune that strain after rooting, if it is an older strain. I wrote [this root and prune script](https://github.com/blab/cartography/blob/master/notebooks/scripts/root_and_prune_tree.py) to accomplish this effect for a different project.

But, this all comes back to @trvrb's [issue about providing an option to root and prune the outgroup in Augur](https://github.com/nextstrain/augur/issues/340) which we should probably just implement soon...